### ISSUE TITLE
Loading and listing NuGet dependencies fail when NuGet package does not simply export a namespace named as the package itself

### ISSUE THREAD

**Issue by Alexander-Block** [aplteam/Cider#58]
This issue arises whenever Cider tries to handle NuGet packages that satisfy one or both of the following two conditions: 
1. their dotnet assembly (i.e. dll file) does not have a filename that is equal to the package itself; or
2. the package does not simply export a namespaces named as itself. 

An example for this is the NuGet package [Parquet.Net](https://www.nuget.org/packages/Parquet.Net) that at the root of its namespace hierarchy exports the two namespaces `Parquet` and `NetBox`. 

An easy way to reproduce this problem is as follows:
```APL
      ]Cider.CreateProject path/to/project
<snip>
      ]Cider.AddNuGetDependencies Parquet.Net
<snip>
      ]Cider.ListNuGetDependencies
* Command Execution Failed: INDEX ERROR
```

In a new session we get the following error:
```APL
      ]Cider.OpenProject path/to/project
* Command Execution Failed: INDEX ERROR
```

In both failure cases the `INDEX ERROR` occurs in the function `GetRealNuGetNames`. This function is meant to normalise names of NuGet dependencies using the names of their dotnet assemblies. Since the dotnet assembly for `Parquet.Net` is named `Parquet.dll` and _not_ `Parquet.Net.dll` this fails, since `GetRealNuGetNames` is not able to relate `Parquet.dll` to `Parquet.Net`.  This is the reason why condition 1 above is a problem.

However, even if we short-circuit this name normalisation we get into the next problem. The function `LoadNuGetDependencies` that is called by `OpenProject` uses this list of names as the list of dotnet namespaces that should be made accessible in the Cider project. This fails again in the working example, since the package `Parquet.Net` does not export a namespace `Parquet.Net`. More subtly it does export a namespace `NetBox` with no relationship whatsoever to the package name which is currently completely ignored. 

In summary the problem seems to be that when handling NuGet dependencies Cider makes the assumption that the following three are always equal:
1. The name of a NuGet package;
2. The file name of the assembly for the NuGet package;
3. The single namespace exported by the NuGet package at the root of its namespace hierarchy. 

These assumptions are not correct in general. To solve this issue it is necessary to properly distinguish between these three types of names in Cider.
